### ISSUE TITLE
Is ssh jump possible?

### ISSUE THREAD

**Issue by Sucharek233** [TerminalStudio/dartssh2#74]
Is using ssh jump possible? Maybe somehow with a proxy?

**Comment by xtyxtyx** [comment]
Hi! Can you provide more information about what you mean by “ssh jump”? Are you trying to use an SSH tunnel to access a remote server through a proxy? Knowing more about your specific use case will help me give you a more helpful response.

**Comment by Sucharek233** [comment]
Hi, thanks for your answer. Yes, I'm using ssh-j.com to connect to a remote server.

**Comment by Sucharek233** [comment]
Hi @xtyxtyx, do you have an answer? Is connecting to a remote ssh server hosted on ssh-j.com possible?

**Comment by xtyxtyx** [comment]
Sorry for the delayed reply. It's totally possible to implement ssh jump with dartssh2 and I've make some tweaks to the library in `2.9.0-pre` to make it even easier to do this. For more detail you can refer to the newly added [example](https://github.com/TerminalStudio/dartssh2/blob/master/example/ssh_jump.dart).

**Comment by Sucharek233** [comment]
Hi, it's giving me an error.
`The argument type 'SSHForwardChannel' can't be assigned to the parameter type 'SSHSocket'.`

The error is located here:
```
  final client = SSHClient(
    await jumpServer.forwardLocal('<target server>', 22), <------
    username: '...',
    onPasswordRequest: () => '...',
  );
```

How can I fix it (my flutter verison is 3.7.7 stable if that helps)?

**Comment by Sucharek233** [comment]
Turns out I didn't update dartssh2 to the newest version (`2.9.0-pre`).
It works perfectly now. Thanks!
